packages/compartment-mapper/src/archive.js

```diff
@@ -134,15 +127,12 @@ const renameSources = (sources, renames) => {
 const addSourcesToArchive = async (archive, sources) => {
   for (const compartment of keys(sources).sort()) {
     const modules = sources[compartment];
-    const compartmentLocation = resolveLocation(`${compartment}/`, 'file:///');
     for (const specifier of keys(modules).sort()) {
       const { bytes, location } = modules[specifier];
       if (location !== undefined) {
-        const moduleLocation = resolveLocation(location, compartmentLocation);
-        const path = new URL(moduleLocation).pathname.slice(1); // elide initial "/"
         if (bytes !== undefined) {
           // eslint-disable-next-line no-await-in-loop
-          await archive.write(path, bytes);
+          await archive.write(`${compartment}/${location}`, bytes);
```

Comment:
location can't do `../` or the like? Where is that specified? looking...

Reply:
it's not constrained beyond `string` in https://github.com/endojs/endo/blob/c1407931517e09c1948e3628d01ba8049ef09318/packages/compartment-mapper/src/types.js#L244